### PR TITLE
Remove 'must not be null' comments from various libraries.

### PR DESCRIPTION
**Pull request by gspencergoog**
## Description

This removes all of the comments that are of the form "so-and-so (must not be null|can ?not be null|must be non-null)" from the cases where those values are defines as non-nullable values.

This PR removes them from the animation, cupertino, foundation, gestures, semantics, and services libraries.  Each of them only had a few, so I lumped them together.

This was done by hand, since it really didn't lend itself to scripting, so it needs to be more than just spot-checked, I think. I was careful to leave any comment that referred to parameters that were nullable, but I may have missed some.

In addition to being no longer relevant after null safety has been made the default, these comments were largely fragile, in that it was easy for them to get out of date, and not be accurate anymore anyhow.

This did create a number of constructor comments which basically say "Creates a [Foo].", but I don't really know how to avoid that in a large scale change, since there's not much you can really say in a lot of cases.  I think we might consider some leniency for constructors to the "Comment must be meaningful" style guidance (which we de facto have already, since there are a bunch of these).

## Related PRs
- https://github.com/flutter/flutter/pull/134991
- https://github.com/flutter/flutter/pull/134992
- https://github.com/flutter/flutter/pull/134993
- https://github.com/flutter/flutter/pull/134994

## Tests
 - Documentation only change.

### CODE REVIEW

File: packages/flutter/lib/src/animation/animations.dart [hunk 1]
```diff
@@ -631,8 +627,9 @@ class TrainHoppingAnimation extends Animation<double>
 /// animation otherwise.
 abstract class CompoundAnimation<T> extends Animation<T>
   with AnimationLazyListenerMixin, AnimationLocalListenersMixin, AnimationLocalStatusListenersMixin {
-  /// Creates a CompoundAnimation. Both arguments must be non-null. Either can
-  /// be a CompoundAnimation itself to combine multiple animations.
+  /// Creates a CompoundAnimation.
+  ///
+  /// Either can be a CompoundAnimation itself to combine multiple animations.
```
Comment (goderbauer) [inline packages/flutter/lib/src/animation/animations.dart:632]:
```suggestion
  /// Either argument can be a CompoundAnimation itself to combine multiple animations.
```

File: packages/flutter/lib/src/cupertino/picker.dart [hunk 2]
```diff
@@ -134,7 +133,7 @@ class CupertinoPicker extends StatefulWidget {
   ///
   /// For more details, see [ListWheelScrollView.diameterRatio].
   ///
-  /// Must not be null and defaults to `1.1` to visually mimic iOS.
+  /// Defaults to `1.1` to visually mimic iOS.
```
Comment (goderbauer) [inline packages/flutter/lib/src/cupertino/picker.dart:136]:
```suggestion
  /// Defaults to 1.1 to visually mimic iOS.
```

File: packages/flutter/lib/src/cupertino/sliding_segmented_control.dart [hunk 3]
```diff
@@ -403,7 +401,7 @@ class CupertinoSlidingSegmentedControl<T> extends StatefulWidget {
 
   /// The amount of space by which to inset the [children].
   ///
-  /// Must not be null. Defaults to EdgeInsets.symmetric(vertical: 2, horizontal: 3).
+  /// Defaults to EdgeInsets.symmetric(vertical: 2, horizontal: 3).
```
Comment (goderbauer) [inline packages/flutter/lib/src/cupertino/sliding_segmented_control.dart:404]:
```suggestion
  /// Defaults to `EdgeInsets.symmetric(vertical: 2, horizontal: 3)`.
```

File: packages/flutter/lib/src/cupertino/switch.dart [hunk 4]
```diff
@@ -63,9 +63,6 @@ import 'thumb_painter.dart';
 ///  * <https://developer.apple.com/ios/human-interface-guidelines/controls/switches/>
 class CupertinoSwitch extends StatefulWidget {
   /// Creates an iOS-style switch.
-  ///
-  /// The [value] parameter must not be null.
-  /// The [dragStartBehavior] parameter defaults to [DragStartBehavior.start] and must not be null.
```
Comment (goderbauer) [inline packages/flutter/lib/src/cupertino/switch.dart:68]: Maybe keep the note about the default value?

File: packages/flutter/lib/src/services/platform_views.dart [hunk 5]
```diff
@@ -92,28 +92,33 @@ class PlatformViewsService {
   /// {@template flutter.services.PlatformViewsService.initAndroidView}
   /// Creates a controller for a new Android view.
   ///
-  /// `id` is an unused unique identifier generated with [platformViewsRegistry].
+  /// `id` is an unused unique identifier generated with
```
Comment (goderbauer) [inline packages/flutter/lib/src/services/platform_views.dart:95]: Here and elsewhere in this file: According to style guide sentences should start grammatically correct with a capital letter. May have to reword the sentence to not start with the parameter name.

Comment (goderbauer) [reply packages/flutter/lib/src/services/platform_views.dart:95]: (I realize this is an existing problem, so feel free to punt given the size of this change)

File: packages/flutter/lib/src/services/platform_views.dart [hunk 6]
```diff
@@ -196,19 +201,22 @@ class PlatformViewsService {
     return controller;
   }
 
-  // TODO(amirh): reference the iOS plugin API for registering a UIView factory once it lands.
-  /// This is work in progress, not yet ready to be used, and requires a custom engine build. Creates a controller for a new iOS UIView.
+  // TODO(amirh): reference the iOS plugin API for registering a UIView factory
+  // once it lands.
+
+  /// This is work in progress, not yet ready to be used, and requires a custom
```
Comment (goderbauer) [inline packages/flutter/lib/src/services/platform_views.dart:207]: (unrelated, but maybe we can update) This feels horribly out of date. Maybe somebody from the iOS side can advise on how to update this to more accurately reflect reality? (cc @stuartmorgan)

Comment (goderbauer) [reply packages/flutter/lib/src/services/platform_views.dart:207]: cc @cbracken

File: packages/flutter/lib/src/cupertino/switch.dart [hunk 7]
```diff
@@ -63,6 +63,9 @@ import 'thumb_painter.dart';
 ///  * <https://developer.apple.com/ios/human-interface-guidelines/controls/switches/>
 class CupertinoSwitch extends StatefulWidget {
   /// Creates an iOS-style switch.
+  ///
+  /// The [dragStartBehavior] parameter defaults to [DragStartBehavior.start].
+
```
Comment (goderbauer) [inline packages/flutter/lib/src/cupertino/switch.dart:68]: nit: remove empty line?

File: packages/flutter/lib/src/services/platform_views.dart (hunk 6 above)
Comment (cbracken) [reply packages/flutter/lib/src/services/platform_views.dart:207]: Thanks! I was looking for this exact comment about a week ago! Spotted it about 6 months ago and decided to leave this till we'd decided on what the final UiKitView vs AppKitView API looked like, then went to fix it the other day and couldn't remember the exact wording or location (was looking on the UiKitView class and related render view etc. themselves).

I'll send a quick patch to rewrite this consistently and also fix the TODO while I'm in there.

Comment (cbracken) [reply packages/flutter/lib/src/services/platform_views.dart:207]: Feel free to land this cleanup first though; probably easier to deal with than the merge conflict from my rewrite!

File: packages/flutter/lib/src/cupertino/switch.dart (hunk 7 above)
Comment (gspencergoog) [reply packages/flutter/lib/src/cupertino/switch.dart:68]: Done.

File: packages/flutter/lib/src/services/platform_views.dart (hunk 6 above)
Comment (gspencergoog) [reply packages/flutter/lib/src/services/platform_views.dart:207]: Thanks Chris!

Comment (gspencergoog) [reply packages/flutter/lib/src/services/platform_views.dart:207]: Fixed in https://github.com/flutter/flutter/pull/135040

**Review by goderbauer — APPROVED** [review]
LGTM

**Pull request merged**